src/dtl/flux_dtl.c

```diff
@@ -0,0 +1,144 @@
+#include "flux_dtl.h"
+
+dyad_rc_t dyad_dtl_flux_init (dyad_dtl_flux_t** dtl_handle,
+        dyad_dtl_mode_t mode, flux_t* h, bool debug)
+{
+    *dtl_handle = malloc(sizeof(struct dyad_dtl_flux));
+    if (*dtl_handle == NULL) {
+        FLUX_LOG_ERR (h, "Cannot allocate the Flux DTL handle\n");
+        return DYAD_RC_SYSFAIL;
+    }
+    (*dtl_handle)->h = h;
+    (*dtl_handle)->debug = debug;
+    (*dtl_handle)->f = NULL;
+    (*dtl_handle)->msg = NULL;
+    return DYAD_RC_OK;
+}
+
+dyad_rc_t dyad_dtl_flux_rpc_pack (dyad_dtl_t* self, const char* upath,
+        uint32_t producer_rank, json_t** packed_obj)
+{
+    dyad_dtl_flux_t* dtl_handle = self->private.flux_dtl_handle;
+    *packed_obj = json_pack(
+        "{s:s}",
+        "upath",
+        upath
+    );
+    if (*packed_obj == NULL) {
+        FLUX_LOG_ERR (dtl_handle->h, "Could not pack upath for Flux DTL\n");
+        return DYAD_RC_BADPACK;
+    }
+    return DYAD_RC_OK;
+}
+
+dyad_rc_t dyad_dtl_flux_rpc_unpack (dyad_dtl_t* self,
+        const flux_msg_t* msg, char** upath)
+{
+    int errcode = 0;
+    errcode = flux_request_unpack (
+        msg,
+        NULL,
+        "{s:s}",
+        "upath",
+        upath
+    );
+    if (errcode < 0) {
```

Comment:
TODO: Check this with a macro as it is done some other places. Also, rc is better than errorcode.

Reply:
There are similar lines.

Reply:
This check should **not** use `DYAD_IS_ERROR` because it's not checking a DYAD return code. It's checking a Flux return code. The choice of variable name was also intentional to emphasize this difference.

Reply:
It does not need to be DYAD_IS_ERROR. It can be FLUX_IS_ERROR for example. error code is misleading because it is not necessarily a code for error like errorno.

Reply:
Just pushed a new commit that deals with this.